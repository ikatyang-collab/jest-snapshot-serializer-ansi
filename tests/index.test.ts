import { expect, test } from 'vitest'
import serializer from '../src/index.js'

expect.addSnapshotSerializer(serializer)

test('ansi', () => {
  const text = 'text'
  const red = `\x1B[31m${text}\x1B[39m`
  const lowerCased = `\x1b[31m${text}\x1b[39m`
  const blue = `\x1B[34m${text}\x1B[39m`

  expect(red).not.toEqual(text)
  expect(text).toMatchInlineSnapshot(`"text"`)
  expect(red).toMatchInlineSnapshot(`text`)
  expect(lowerCased).toMatchInlineSnapshot(`text`)
  expect(blue).toMatchInlineSnapshot(`text`)

  expect(new TypeError(red)).toMatchInlineSnapshot(`[TypeError: text]`)
  expect(() => {
    throw new TypeError(red)
  }).toThrowErrorMatchingInlineSnapshot(`[TypeError: text]`)
  expect(new TypeError(red)).toMatchSnapshot()
})
