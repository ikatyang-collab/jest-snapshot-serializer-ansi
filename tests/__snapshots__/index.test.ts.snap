// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ansi 7`] = `[TypeError: text]`;
